lib/libxdp/libxdp.c

```diff
@@ -3223,6 +3223,50 @@ static int xdp_multiprog__unpin(struct xdp_multiprog *mp)
 	return err;
 }
 
+static int xdp_detach_link(__u32 ifindex, __u32 prog_id) {
+	struct bpf_link_info link_info;
+	__u32 link_info_len, id = 0;
+	int err, fd;
+
+	while (true) {
+		err = bpf_link_get_next_id(id, &id);
+		if (err) {
+			err = -errno;
+			pr_debug("can't get next link: %s (id %d)", strerror(errno), id);
```

Comment:
Lets make the order of id and error message consistent across the different debug prints. We also generally capitalise sentences in debug output throughout libxdp...